airflow/contrib/hooks/gcs_hook.py

```diff
@@ -175,7 +175,30 @@ def download(self, bucket_name, object_name, filename=None):
 
         return blob.download_as_string()
 
-    def upload(self, bucket_name, object_name, filename,
```

Comment:
Is this going to break the operators that are calling gcs_hook.upload()?

Reply:
Yes it would because the new version would replace 'upload' with 'upload_from_filename' and 'upload_from_string'. I can try adding a 'data' (string) and 'from_string' (Boolean) parameter to the 'upload' method, however it is difficult because then the method still has the filename parameter.